### PR TITLE
feat(tests): Init test toolkit documentation

### PR DESCRIPTION
**Pull request by alachambre**

### CODE REVIEW

File: modules/process/pages/process-integration-test.adoc [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+If the field `hungry` is set to `true`, then the path to the task `goToHunt` is selected, else its the path to the task `goToSleep`.
+
+[#quick-start-test]
+=== Create a Bonita test project
+
+Create a new Maven project with the following pom.xml: 
+
+[source,xml]
+----
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (rbioteau) [inline modules/process/pages/process-integration-test.adoc:62]: some dependencies are not required and some plugins are not configured

**Review by rbioteau — CHANGES_REQUESTED** [review]

File: modules/process/pages/process-integration-test.adoc [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+    @RegisterExtension
+    static BonitaTestExtension bonitaExtension = new BonitaTestExtension(Configuration.builder()
+            .targetRuntimeURL(() -> "http://localhost:8080/bonita")
+            .deleteProcessInstances()
+            .clearBDM()
+            .build()); <1>
+
+    @Test
+    public void should_create_a_tyrannosaurus(BonitaTestToolkit toolkit) throws Exception { <2>
```
Comment (alachambre) [inline modules/process/pages/process-integration-test.adoc]:
```suggestion
    public void should_create_an_hungry_tyrannosaurus(BonitaTestToolkit toolkit) throws Exception { <2>
```

File: modules/process/pages/process-integration-test.adoc [hunk 3]
```diff
@@ -0,0 +1,234 @@
+= Create process integration tests
+:description: This pages describes how to write process integration tests using the Bonita Test Toolkit.
+
+{description}
+
+[NOTE]
+====
+The Bonita Test Toolkit is only available for Enterprise, Performance, Efficiency and teamwork editions only. 
+
+However, it based on the open source https://github.com/bonitasoft/bonita-java-client[Bonita Java Client] to make HTTP calls on the Bonita platform. Any user can make its own test toolkit using this client.
```
Comment (educhastenier) [inline modules/process/pages/process-integration-test.adoc]:
```suggestion
However, it is based on the open source https://github.com/bonitasoft/bonita-java-client[Bonita Java Client] to make HTTP calls to the Bonita runtime. Any user can make their own test toolkit using this client. 
```

File: modules/process/pages/process-integration-test.adoc [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+The Bonita Test Toolkit is only available for Enterprise, Performance, Efficiency and teamwork editions only. 
+
+However, it based on the open source https://github.com/bonitasoft/bonita-java-client[Bonita Java Client] to make HTTP calls on the Bonita platform. Any user can make its own test toolkit using this client. 
+====
+
+Some developer skills are required to follow this documentation page and, generally, to write process integration tests.
+
+== Motivations
+
+In the world of development, writing tests is not an option. It is the best way to ensure that a product fulfill its requirements, and that existing existing features won't be broken by future developments.
```
Comment (educhastenier) [inline modules/process/pages/process-integration-test.adoc]:
```suggestion
In the world of development, writing tests is not an option. It is the best way to ensure that a product fulfill its requirements, and that existing features won't be broken by future developments. 
```

File: modules/process/pages/process-integration-test.adoc [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+However, it based on the open source https://github.com/bonitasoft/bonita-java-client[Bonita Java Client] to make HTTP calls on the Bonita platform. Any user can make its own test toolkit using this client. 
+====
+
+Some developer skills are required to follow this documentation page and, generally, to write process integration tests.
+
+== Motivations
+
+In the world of development, writing tests is not an option. It is the best way to ensure that a product fulfill its requirements, and that existing existing features won't be broken by future developments. 
+
+This apply to Bonita processes. For each process, an integration test should be written, to validate that the process executes correctly depending on the inputs, that after each task the platform is in the expected state and so on.
```
Comment (educhastenier) [inline modules/process/pages/process-integration-test.adoc]:
```suggestion
This applies to Bonita processes. For each process, an integration test should be written, to validate that the process executes correctly depending on the inputs, that after each task the platform is in the expected state and so on.
```

File: modules/process/pages/process-integration-test.adoc [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+    public void should_create_an_hungry_tyrannosaurus(BonitaTestToolkit toolkit) throws Exception { <2>
+        var user = toolkit.getUser("walter.bates"); <3>
+        var processDef = toolkit.getProcessDefinition("create-dinosaur"); <4>
+        var businessObject = toolkit.getBusinessOject("com.company.model.Dinosaur"); <5>
+
+        assertThat(businessObject.findAll(0, 10)).isEmpty();
+
+        var processInstance = processDef.startProcessFor(user); <6>
+        BonitaAssertions.assertThat(processInstance).isStarted();
+        BonitaAssertions.assertThat(processInstance).containsPendingTasks("CreateDinosaur"); <7>
```
Comment (benjaminParisel) [inline modules/process/pages/process-integration-test.adoc]:
```suggestion
        BonitaAssertions.assertThat(processInstance).containsPendingUserTasks("CreateDinosaur"); <7>
```

File: modules/process/pages/process-integration-test.adoc [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-failsafe-plugin</artifactId>
+                    <version>${maven-failsafe-plugin.version}</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+        
+        <plugins>
+            <groupId>org.apache.maven.plugins</groupId>
```
Comment (alachambre) [inline modules/process/pages/process-integration-test.adoc:140]:
```suggestion
<plugin>
            <groupId>org.apache.maven.plugins</groupId>
```

File: modules/process/pages/process-integration-test.adoc [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+            <artifactId>maven-failsafe-plugin</artifactId>
+            <executions>
+                <execution>
+                    <goals>
+                        <goal>integration-test</goal>
+                        <goal>verify</goal>
+                    </goals>
+                </execution>
+            </executions>
+        </plugins>
```
Comment (alachambre) [inline modules/process/pages/process-integration-test.adoc:151]:
```suggestion
	</plugin>
        </plugins>
```

**Pull request merged**